BloomFramework/src/Input/InputManager.cpp

```diff
@@ -0,0 +1,215 @@
+				m_mouseY = event.motion.y;
+				m_mouseMoveX = event.motion.xrel;
+				m_mouseMoveY = event.motion.yrel;
+				break;
+			}
+			case SDL_MOUSEBUTTONDOWN: {
+				m_mouse = SDL_GetMouseState(&(m_mouseX),
+					&(m_mouseY));
+
+				m_mouseDown[event.button.button] = true;
+				break;
+			}
+			case SDL_MOUSEBUTTONUP: {
+				m_mouse = SDL_GetMouseState(&(m_mouseX),
+					&(m_mouseY));
+
+				m_mouseDown[event.button.button] = true;
+				break;
+			}
+			case SDL_MOUSEWHEEL: {
+				m_scrollX = event.wheel.x;
+				m_scrollY = event.wheel.y;
+				break;
+			}
+			default:
+				break;
+			}
+		}
+	}
+
+	bool InputManager::isKeyDown(int key) {
+		if (m_isLocked) return false;
+
+		if (key < 0 || key >= KEYBOARD_SIZE)
+			return false;
+
+		return (m_keyDown[key]);
+	}
+
+	bool InputManager::isKeyUp(int key) {
+		if (m_isLocked) return false;
+
+		if (key < 0 || key >= KEYBOARD_SIZE)
+			return false;
+
+		return (m_keyUp[key]);
+	}
+
+	bool InputManager::isKeyPressed(KeyboardKey key) {
+		if (m_isLocked) return false;
+
+		if (!(m_keyboard))
+			return false;
+
+		if (m_keyboard[key])
+			return true;
+
+		return false;
+	}
+
+	bool InputManager::shift() {
+		return (isKeyPressed(KEY_LEFT_SHIFT) || isKeyPressed(KEY_RIGHT_SHIFT));
+	}
+
+	bool InputManager::ctrl() {
+		return (isKeyPressed(KEY_LEFT_CTRL) || isKeyPressed(KEY_RIGHT_CTRL));
+	}
+
+	bool InputManager::alt() {
+		return (isKeyPressed(KEY_LEFT_ALT) || isKeyPressed(KEY_RIGHT_ALT));
+	}
+
+	bool InputManager::isMouseDown(MouseButton button) {
+		if (m_isLocked) return false;
+
+		if (button == MOUSE_MAX)
+			return false;
+
+		return m_mouseDown[button];
+	}
+
+	bool InputManager::isMouseUp(MouseButton button) {
+		if (m_isLocked) return false;
+
+		if (button == MOUSE_MAX)
+			return false;
+
+		return m_mouseUp[button];
+	}
+
+	bool InputManager::isMousePressed(MouseButton button) {
+		if (m_isLocked) return false;
+
+		switch (button)
+		{
+		case MOUSE_LEFT:
+			if (m_mouse & SDL_BUTTON(1))
+				return true;
+			break;
+
+		case MOUSE_MIDDLE:
+			if (m_mouse & SDL_BUTTON(2))
+				return true;
+			break;
+
+		case MOUSE_RIGHT:
+			if (m_mouse & SDL_BUTTON(3))
+				return true;
+			break;
+
+		default:
+			break;
+		}
+
+		return false;
+	}
+
+	int InputManager::getMouseX() {
+		return m_mouseX;
+	}
+
+	int InputManager::getMouseY() {
+		return m_mouseY;
+	}
+
+	bool InputManager::quitRequested() {
+		bool curr = m_will_quit;
+		m_will_quit = false;
+		return curr;
+	}
+
+	bool InputManager::isMouseInside(SDL_Rect rectangle) {
+		if ((m_mouseX >= rectangle.x) 
+			&& (m_mouseX <= rectangle.x + rectangle.w)
+			&& (m_mouseY >= rectangle.y) 
+			&& (m_mouseY <= rectangle.y + rectangle.h))
+			return true;
+
+		return false;
+	}
+
+	bool InputManager::isPrintable(SDL_Keycode key) {
+		return ((key > SDLK_SPACE) && (key < SDLK_z));
+	}
+
+	bool InputManager::isPrintableKeyDown() {
+		return (InputManager::isPrintable(m_curPrintableKey));
+	}
+
+	std::string InputManager::getCurPrintableKey() {
```

Comment:
what the point of using `string` here?

Reply:
Using '\0' char in `std::cout` will output a space instead of nothing.